Mathlib/Tactic/Finiteness.lean

```diff
@@ -0,0 +1,36 @@
+/-
+Copyright (c) 2023 Heather Macbeth. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Heather Macbeth
+-/
+import Mathlib.Tactic.Positivity.Core
+
+/-! # Finiteness tactic
+
+This file implements a basic `finiteness` tactic, designed to solve goals of the form `*** < ∞` and
+(equivalently) `*** ≠ ∞` in the extended nonnegative reals (`ENNReal`, aka `ℝ≥0∞`).
+
+It works recursively according to the syntax of the expression. It is implemented as an `aesop` rule
+set.
+
+[TODO]: improve `finiteness` to also deal with other situations, such as balls in proper spaces with
+a locally finite measure.
+-/
+
+
+open Aesop.BuiltinRules in
+attribute [aesop (rule_sets := [finiteness]) safe -50] assumption intros
+
+open Lean Elab Tactic in
+/-- A version of the positivity tactic for use by `aesop`. -/
+@[aesop safe tactic (rule_sets := [finiteness])]
+def PositivityForAesop : TacticM Unit :=
+  liftMetaTactic fun g => do Mathlib.Meta.Positivity.positivity g; pure []
```

Comment:
I have since learned that there is a better syntax for this, something like (untested)
```suggestion
  liftMetaFinishingTactic Mathlib.Meta.Positivity.positivity
```